### PR TITLE
Always show supporter-only playlist durations

### PR DESCRIPTION
**Pull request by Joehuu**
osu!_7fvQPtqQuX

### CODE REVIEW

File: osu.Game/Localisation/OnlinePlayStrings.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+namespace osu.Game.Localisation
+{
+    public static class OnlinePlayStrings
+    {
+        private const string prefix = @"osu.Game.Resources.Localisation.OnlinePlay";
+
+        /// <summary>
+        /// "This duration is only available for osu!supporters."
+        /// </summary>
+        public static LocalisableString SupporterOnlyDurationNotice => new TranslatableString(getKey(@"supporter_only_duration_notice"), @"This duration is only available for osu!supporters.");
```
Comment (peppy) [inline osu.Game/Localisation/OnlinePlayStrings.cs:15]: Might want to phrase this similar to the other case we already have:

https://github.com/ppy/osu/blob/de7f2a0bba6953b486689427198b0b1fd2804a30/osu.Game/Localisation/LeaderboardStrings.cs#L42-L45

File: osu.Game/Screens/OnlinePlay/Playlists/PlaylistsRoomSettingsOverlay.cs [hunk 2]
```diff
@@ -326,18 +351,9 @@ private void populateDurations(ValueChangedEvent<APIUser> user)
                     TimeSpan.FromDays(3),
                     TimeSpan.FromDays(7),
                     TimeSpan.FromDays(14),
+                    TimeSpan.FromDays(days_in_month),
+                    TimeSpan.FromDays(days_in_month * 3),
```
Comment (peppy) [inline osu.Game/Screens/OnlinePlay/Playlists/PlaylistsRoomSettingsOverlay.cs:355]: I'd hope these had some hinting in the dropdown itself, whether that be a `*` next to it or ` (supporter)` or something else.

File: osu.Game/Screens/OnlinePlay/Playlists/PlaylistsRoomSettingsOverlay.cs [hunk 3]
```diff
@@ -352,7 +368,10 @@ protected override void Update()
             private void onPlaylistChanged(object? sender, NotifyCollectionChangedEventArgs e) =>
                 playlistLength.Text = $"Length: {Playlist.GetTotalDuration()}";
 
-            private bool hasValidSettings => RoomID.Value == null && NameField.Text.Length > 0 && Playlist.Count > 0;
+            private bool hasValidSettings => RoomID.Value == null && NameField.Text.Length > 0 && Playlist.Count > 0
+                                             && selectedAvailableDuration;
+
+            private bool selectedAvailableDuration => DurationField.Current.Value < TimeSpan.FromDays(31) || localUser.Value.IsSupporter;
```
Comment (peppy) [inline osu.Game/Screens/OnlinePlay/Playlists/PlaylistsRoomSettingsOverlay.cs:374]: Rename to `hasValidDuration`.

File: osu.Game/Localisation/OnlinePlayStrings.cs (hunk 1 above)
Comment (Joehuu) [reply osu.Game/Localisation/OnlinePlayStrings.cs:15]: Maybe similar to the one in beatmap listing?
osu!_6opU0y8Ola

`This duration requires an active osu!supporter tag.`

File: osu.Game/Screens/OnlinePlay/Playlists/PlaylistsRoomSettingsOverlay.cs (hunk 2 above)
Comment (Joehuu) [reply osu.Game/Screens/OnlinePlay/Playlists/PlaylistsRoomSettingsOverlay.cs:355]: Other supporter features never hint until it's clicked (i.e. beatmap listing filters and country/friend/mod leaderboards) so those need to be reconsidered too.

**Review by peppy — APPROVED** [review]
Probably okay

**Pull request merged**